### ISSUE TITLE
ochttp should provide a way to pass route info

### ISSUE THREAD

**Issue by rozaliev** [census-instrumentation/opencensus-go#663]
According to the [spec](https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/HTTP.md) the path should be used as a span name only if route can't be determined.

I haven't found any way pass route name,  which is unfortunate for REST-like apis. 

I imagine something like this for api: 
```go
// ochttp.Handler
type Handler struct {
   ...
   RouteName string
   ...
}
```

**Comment by rakyll** [comment]
I wished that mux had a way to report the matched route. Putting this on the Handler as a field might not be ideal because it will require to initiate a new Handler for each route even though our primary recommendation is to wrap the entire mux with it.

**Comment by ramonza** [comment]
Maybe what we need is a way to override the Span name of an active Span:

    trace.FromContext(req.Context()).SetName("...")

**Comment by basvanbeek** [comment]
Gorilla's Mux allows for pulling route information from Request Context. Here's an example of a Go kit serverBefore handler using this i.c.m. zipkin-go and overriding the default span name.

```go
func ZipkinHTTPRoute(ctx context.Context, r *http.Request) context.Context {
	if span := zipkin.SpanFromContext(ctx); span != nil {
		if route := mux.CurrentRoute(r); route != nil {
			if tpl, err := route.GetPathTemplate(); err == nil {
				zipkin.TagHTTPRoute.Set(span, tpl)
				span.SetName(r.Method + " " + tpl)
			}
		}
	}
	return ctx
}
```

**Comment by ramonza** [comment]
+1 for FormatSpanName callback in Handler

**Comment by ramonza** [comment]
Fixed by #796
